subsys/audio_modules/audio_module_template/audio_module_template.c

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright(c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include "audio_module_template.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <ctype.h>
+#include <zephyr/kernel.h>
+#include <zephyr/shell/shell.h>
+#include <errno.h>
+#include "audio_defines.h"
+#include "audio_module.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(audio_module_template, CONFIG_AUDIO_MODULE_TEMPLATE_LOG_LEVEL);
+
+static int audio_module_template_open(struct audio_module_handle_private *handle,
+				      struct audio_module_configuration const *const configuration)
+
+{
+	struct audio_module_handle *hdl = (struct audio_module_handle *)handle;
```

Comment:
Please remind me. Why do we need both a private and a public version?

Reply:
The public is to allow a single API to call many different modules. Within the module this public pointer is cast to the private structure and used only within the module.